### PR TITLE
Report course_key and certificate_type to cybersource.

### PR DESCRIPTION
**Pull request by jimabramson**
XCOM-274

@rlucioni or @clintonb please review.

### CODE REVIEW

File: ecommerce/extensions/payment/processors.py [hunk 1]
```diff
@@ -158,6 +165,20 @@ def get_transaction_parameters(self, basket):
...
+        """
+        Return the first product encountered in the basket with the product
+        class of 'seat'.  Return None if no such products were found.
+        """
+        try:
+            seat_class = ProductClass.objects.get(slug='seat')
+        except ProductClass.DoesNotExist:
+            # this occurs in test configurations where the seat product class is not in use
+            return None
+        products = [line.product for line in basket.lines.all() if line.product.product_class == seat_class]
```
Comment (clintonb) [inline ecommerce/extensions/payment/processors.py]: Minor (not sure of performance implications): You can do the filtering the DB: 

``` python
line = basket.lines.filter(product__product_class=seat_class).first()
return line.product if line else None
```

File: ecommerce/extensions/payment/tests/test_processors.py [hunk 2]
```diff
@@ -31,10 +31,26 @@ class PaymentProcessorTestCaseMixin(PaymentEventsMixin):
...
 
+    COURSE_KEY = 'test-course-key'
+    CERTIFICATE_TYPE = 'test-certificate-type'
+
     def setUp(self):
         super(PaymentProcessorTestCaseMixin, self).setUp()
+
+        # certain logic and tests expect to find products with class 'seat' in
+        # baskets, so that's done explicitly in this setup.
+        self.product_class = factories.ProductClassFactory(slug='seat', requires_shipping=False, track_stock=False)
```
Comment (clintonb) [inline ecommerce/extensions/payment/tests/test_processors.py:42]: This should call save automatically.

File: ecommerce/extensions/payment/processors.py (hunk 1 above)
Comment (jimabramson) [inline ecommerce/extensions/payment/processors.py]: nice, using.

File: ecommerce/extensions/payment/tests/test_processors.py (hunk 2 above)
Comment (jimabramson) [inline ecommerce/extensions/payment/tests/test_processors.py:42]: good call.   removed all the saves, and it's still happy.

**Comment by jimabramson** [comment]
updated

**Comment by clintonb** [comment]
:+1:

**Pull request merged**